presto-base-jdbc/src/test/java/com/facebook/presto/plugin/jdbc/TestJdbcQueryBuilder.java

```diff
@@ -123,6 +130,52 @@ public void testNormalBuildSql()
     }
 
     @Test
+    public void testBuildSqlWithString()
+            throws SQLException
+    {
+        TupleDomain<ColumnHandle> tupleDomain = TupleDomain.withColumnDomains(ImmutableMap.of(
+                cols.get(0), Domain.create(SortedRangeSet.copyOf(BigintType.BIGINT,
+                        ImmutableList.of(
+                                Range.equal(BigintType.BIGINT, 128L),
+                                Range.equal(BigintType.BIGINT, 180L),
+                                Range.equal(BigintType.BIGINT, 233L),
+                                Range.lessThan(BigintType.BIGINT, 25L),
+                                Range.range(BigintType.BIGINT, 66L, true, 96L, true),
+                                Range.greaterThan(BigintType.BIGINT, 192L))),
+                        false),
+                cols.get(1), Domain.create(SortedRangeSet.copyOf(DoubleType.DOUBLE,
+                        ImmutableList.of(
+                                Range.equal(DoubleType.DOUBLE, 200011.0),
+                                Range.equal(DoubleType.DOUBLE, 200014.0),
+                                Range.equal(DoubleType.DOUBLE, 200017.0),
+                                Range.equal(DoubleType.DOUBLE, 200116.5),
+                                Range.range(DoubleType.DOUBLE, 200030.0, true, 200036.0, true),
+                                Range.range(DoubleType.DOUBLE, 200048.0, true, 200099.0, true))),
+                        false),
+                cols.get(2), Domain.create(SortedRangeSet.copyOf(BooleanType.BOOLEAN,
+                        ImmutableList.of(Range.equal(BooleanType.BOOLEAN, true))),
+                        false),
+                cols.get(3), Domain.create(SortedRangeSet.copyOf(VarcharType.VARCHAR,
+                        ImmutableList.of(
+                                Range.range(VarcharType.VARCHAR, Slices.wrappedBuffer(ByteBuffer.wrap("test_str_68".getBytes())), true, Slices.wrappedBuffer(ByteBuffer.wrap("test_str_96".getBytes())), false),
+                                Range.equal(VarcharType.VARCHAR, Slices.wrappedBuffer(ByteBuffer.wrap("test_str_180".getBytes()))),
+                                Range.equal(VarcharType.VARCHAR, Slices.wrappedBuffer(ByteBuffer.wrap("test_str_196".getBytes()))))),
+                        false)
+        ));
+        Connection connection = dummyHandle.getConnection();
```

Comment:
All JDBC objects need to be closed, so these should be created using `try-with-resources`. Fix this and all other places